MediaBrowser.Api/BaseApiService.cs

```diff
@@ -16,30 +18,43 @@ namespace MediaBrowser.Api
     /// <summary>
     /// Class BaseApiService
     /// </summary>
-    public class BaseApiService : IService, IRequiresRequest
+    public abstract class BaseApiService : IService, IRequiresRequest
     {
+        public BaseApiService(
+            ILogger logger,
+            IServerConfigurationManager serverConfigurationManager,
+            IHttpResultFactory httpResultFactory)
+        {
+            Logger = logger;
+            ServerConfigurationManager = serverConfigurationManager;
+            ResultFactory = httpResultFactory;
+        }
+
         /// <summary>
-        /// Gets or sets the logger.
+        /// Gets the logger.
         /// </summary>
         /// <value>The logger.</value>
-        public ILogger Logger => ApiEntryPoint.Instance.Logger;
+        protected ILogger Logger { get; }
```

Comment:
why?

Reply:
actually, I should have asked a more generic question - why turn this class into an `abstract` one, what's the reasoning?

Reply:
It should only be inherited and never created a direct instance of

Reply:
But is that _required_ to be done in this PR? Looks like a general architecture cleanup to me.

Reply:
This logger property change is, the  addition of abstract not.

Reply:
well, if we remove all fields then it doesn't make sense to _not_ make it abstract... so I'll stop objecting here :)